flake.nix

```diff
@@ -51,7 +51,7 @@
       src,
     }:
     # modules/profiles are always functions
-    args: let
+    args @ {pkgs, ...}: let
```

Comment:
I think a module system without `pkgs` is at least conceivable, so we should preemptively do `pkgs ? {}`

Reply:
@whs-dot-hk what's your thought, here?

Reply:
Yes, I think so, with a default `{}` is better

Reply:
That wouldn't work, because `applyModuleArgs` would fail. You cannot specify default arguments for module functions: https://github.com/NixOS/nixpkgs/issues/243303.

Reply:
That wouldn't work, because `applyModuleArgs` would fail. You cannot specify default arguments for module functions: https://github.com/NixOS/nixpkgs/issues/243303.

Reply:
Wow! Thanks for curating the reference here.

Reply:
Nice, can confirm the default do nothing (such as for `pkgs`) with `mytest`, when there is no `config._module.args.mytest = "test";` there will be an error even a default `""` is given (`mytest ? ""`)